src/components/common/QRCodeComponent.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+    <div class="">
+        <q-btn no-caps color="secondary" text-color="black" spread label="Generate QR Code" data-cy="share-button" style="width: 240px;"
+            @click="showQRCodePopup = true">
+        </q-btn>
+        <q-dialog class="q-pa-lg" v-model="showQRCodePopup" @show="onDialogShow">
+            <q-card class="custom-width">
+                <div class="flex row justify-between items-center">
+                    <q-card-section class="text-h6 flex justify-center">
+                        QR Code
+                    </q-card-section>
+                    <q-card-actions align="center" class="justify-center">
+                        <q-btn flat icon="sym_r_close" color="primary" @click="showQRCodePopup = false" />
+                    </q-card-actions>
+                </div>
+
+                <q-card-section class="q-pa-md">
+                    <div class="q-mb-md">
+                        <!-- Render canvas only when dialog is open -->
+                        <canvas v-if="showQRCodePopup" ref="qrCanvas"  class="qr-code-canvas"/>
+                    </div>
+
+                    <q-input v-model="qrLink" label="Link" outlined readonly />
+                    <q-btn label="Copy link" color="primary" class="q-mt-md center" @click="copyToClipboard" />
+
+                    <!-- Share Buttons Section -->
+                    <div class="q-mt-md flex justify-center">
+                            <q-btn icon="fab fa-facebook" color="blue" flat @click="shareOnPlatform('facebook')" />
+                            <q-btn icon="fab fa-twitter" color="blue" flat @click="shareOnPlatform('twitter')" />
+                            <q-btn icon="fab fa-linkedin" color="blue" flat @click="shareOnPlatform('linkedin')" />
+                            <q-btn icon="fab fa-whatsapp" color="green" flat @click="shareOnPlatform('whatsapp')" />
+                            <q-btn icon="fas fa-envelope" color="red" flat @click="shareOnPlatform('email')" />
+
+                    </div>
+                </q-card-section>
+            </q-card>
+        </q-dialog>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { ref, nextTick, onMounted } from 'vue'
+import QRCode from 'qrcode'
+
+const showQRCodePopup = ref(false)
+const qrLink = ref<string>('')
+const qrCanvas = ref<HTMLCanvasElement | null>(null)
+
+onMounted(() => {
+  qrLink.value = window.location.href
+})
+
+const generateQRCode = () => {
+  const canvas = qrCanvas.value
+  if (canvas) {
+    QRCode.toCanvas(canvas, qrLink.value, { width: 200 }, (error) => {
+      if (error) console.error(error)
+    })
+  } else {
+    console.error('Canvas element is not available.')
+  }
+}
+
+// Run QR code generation when dialog shows
+const onDialogShow = () => {
+  nextTick(() => {
+    generateQRCode()
+  })
+}
+
+const copyToClipboard = () => {
+  navigator.clipboard.writeText(qrLink.value).then(() => {
+    alert('Link copied to clipboard!')
```

Comment:
There is `useNotification.ts` composible allowing to display `success()` notification. Can be more attractive then standard alert